monai/apps/pathology/losses/hovernet_loss.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+
+import torch
+from torch.nn import CrossEntropyLoss
+from torch.nn import functional as F
+from torch.nn.modules.loss import _Loss
+
+from monai.losses import DiceLoss
+from monai.transforms import SobelGradients
+from monai.utils.enums import HoVerNetBranch
+
+
+class HoVerNetLoss(_Loss):
+    """
+    Loss function for HoVerNet pipeline, which is combination of losses across three branches
+
+    Args:
+        lambda_hv_mse: Weight factor to apply to the HV regression MSE part of the overall loss
+        lambda_hv_mse_grad: Weight factor to apply to the MSE of the HV gradient part of the overall loss
+        lambda_nuclei_ce: Weight factor to apply to the nuclei prediction CrossEntropyLoss part
+            of the overall loss
+        lambda_nuclei_dice: Weight factor to apply to the nuclei prediction DiceLoss part of overall loss
+        lambda_type_ce: Weight factor to apply to the nuclei class prediction CrossEntropyLoss part
+            of the overall loss
+        lambda_type_dice: Weight factor to apply to the nuclei class prediction DiceLoss part of the
+            overall loss
+    """
+
+    def __init__(
+        self,
+        lambda_hv_mse: float = 2.0,
+        lambda_hv_mse_grad: float = 1.0,
+        lambda_nuclei_ce: float = 1.0,
+        lambda_nuclei_dice: float = 1.0,
+        lambda_type_ce: float = 1.0,
+        lambda_type_dice: float = 1.0,
```

Comment:
Now that we have `HV`,`NP`, and `NC` as they keys for enums of HoVerNet branches. Do you think it is better to make the weighting factors consistent with the branch names and use these names instead, e.g. `lambda_np_ce`, etc.